### ISSUE TITLE
Should `TripolarGrid` really be `RightConnected` in `y`?

### ISSUE THREAD

**Issue by glwagner** [CliMA/OrthogonalSphericalShellGrids.jl#38]
The tripolar grid is abstracted using a `RightConnected` topology in `y`:

https://github.com/CliMA/OrthogonalSphericalShellGrids.jl/blob/379d0de985a0927c026e6c3e43b2ebd163e054aa/src/tripolar_grid.jl#L296

This seems like an abuse. `RightConnected` was originally designed to represent the case that the left side is `Bounded`, while the right side is "open" / "connected" to another domain --- in a similar manner to Periodic. This encapsulates the concept of boundary "ownership"; the present panel does not own the right boundary.

This isn't the case for a tripolar grid. The right boundary in `y` is in fact owned by the tripolar grid and it is not connected to anything else. Instead the boundary is folded onto itself.

This is implemented with a `ZipperBoundaryCondition` but I think this is wrong. It should be implemented with a topology instead. Consider that this condition applies to _all_ fields. It's rather similar in specificity to the `Periodic` case. `RightConnected` is different, it implies that we require knowledge of another domain.

**Comment by glwagner** [comment]
I think we need a `Folded` topology which would then be paired with `Zipper` or `Folded` boundary classification.

**Comment by simone-silvestri** [comment]
Leaving definitions of individual boundaries to topology has the downside of saying something about the _whole_ y-direction while, instead we only want to define one boundary. 
I think that a name like `RightFolded` might be better, because the left boundary can be either distributed or bounded.

**Comment by simone-silvestri** [comment]
Actually, in the case this domain is distributed on the other side it would be `LeftConnected` with a fold on the right

**Comment by glwagner** [comment]
It makes sense to me that the topology depends on whether it is distributed or not. In this issue I am talking only about non-distributed TripolarGrids (also the global grid in a distributed context).

**Comment by glwagner** [comment]
> Leaving definitions of individual boundaries to topology has the downside of saying something about the _whole_ y-direction while, instead we only want to define one boundary.

This is true for sure, but I'm not sure what you are proposing as a solution?
